### ISSUE TITLE
Solution to Python raw strings

### ISSUE THREAD

**Issue by ajitid** [dracula/visual-studio-code#36]
You were right back then about escape characters in Python for raw strings. 
But I found out how other themes are tackling it...

Nord
![screenshot 192](https://cloud.githubusercontent.com/assets/26769888/26021324/a2f2dbca-37a8-11e7-8f9f-3ee2cf84b683.png)

Material theme
![screenshot 193](https://cloud.githubusercontent.com/assets/26769888/26021330/b15bdef0-37a8-11e7-979c-7ec8adfb2fa0.png)

Dracula
![screenshot 191](https://cloud.githubusercontent.com/assets/26769888/26021331/b6379d4c-37a8-11e7-92fd-a06f35f1ef31.png)

Basically they are giving same color to whole raw string the color of escape characters. It seems good to me. Please do it if doesn't breaks the theme. 

That maybe a small thing for anyone, but it brings consistency and sense to semantic highlighting. Thanks!

**Comment by dsifford** [comment]
Thanks @ajitid!

I'm actually doing some theme housekeeping right now so I'll give this a look in a few 😄

**Comment by ajitid** [comment]
I'm in no rush

**Comment by ajitid** [comment]
New syntax highlighting in 2.0 breaks this again and I don't want to create a new issue for this small thing. Please fix it. Thanks. Your work is great to uniform Dracula. Thanks!

**Comment by dsifford** [comment]
@ajitid Yeah, that was intentional.

I did some reading into the Python documentation on this. There's very, very few reasons to ever use the `r"` pattern other than with regular expressions.

According to the docs:

> Raw string notation (r"text") keeps regular expressions sane. Without it, every backslash ('\') in a regular expression would have to be prefixed with another one to escape it. When one wants to match a literal backslash, it must be escaped in the regular expression. With raw string notation, this means r"\\". Without raw string notation, one must use "\\\\"

Finally, the textmate language uses the `r"` exclusively for regular expressions. So, in an effort to make people's lives who depend on regular expressions in python easier, I opted to revert back.

What is your use-case for the `r"` annotation? Why is this affecting you so badly?

**Comment by ajitid** [comment]
If that's for regex then I won't ask you to change it again. 

In Windows for navigation from current path I would do 
`nav = os.getcwd() + r"\new\another\anotherone"` 
which makes it a little clean.

**Comment by ajitid** [comment]
Leave it. Thanks again for updating theme so quickly. I don't know but I think changing syntax highlighting for all languages takes quite much time

**Comment by dsifford** [comment]
Ah, yeah sorry about that. Windows is a drag sometimes..

Another way you could go about it without using `r"` and without escaping the path (`\\new\\another\\anotherone`) would be this...

```py
from os import path, getcwd

nav = path.join(getcwd(), "new", "another", "anotherone")
```

Benefit there would be that it would work across all platforms.

**Comment by lleeoo** [comment]
FTR MagicPython does not consider this a bug: https://github.com/MagicStack/MagicPython#strings. They want to differentiate syntax highlighting of generic raw strings (R'') from regexes (r'').
